File: backend/routers/multi.py
from fastapi import APIRouter, Query

from backend.services.multi_comparison import (
    get_dates as cmp_dates,
    get_comparison_data,
)
from backend.services.multi_interval import (
    get_dates as interval_dates,
    get_interval_data,
)
from backend.services.multi_stats import (
    get_dates as stats_dates,
    get_stats_data,
)

router = APIRouter(tags=["multi"])


# --- Comparison ---
@router.get("/comparison/dates")
def comparison_available_dates(sites: str = Query(...)):
    return {"dates": cmp_dates(sites.split(","))}


@router.get("/comparison/data")
def comparison_data(sites: str = Query(...), date: str = Query(...)):
    return get_comparison_data(sites.split(","), date)


# --- Interval ---
@router.get("/interval/dates")
def interval_available_dates(sites: str = Query(...)):
    return {"dates": interval_dates(sites.split(","))}


@router.get("/interval/data")
def interval_data(
    sites: str = Query(...),
    start_date: str = Query(...),
    end_date: str = Query(...),
):
    return get_interval_data(sites.split(","), start_date, end_date)


# --- Stats ---
@router.get("/stats/dates")
def stats_available_dates(sites: str = Query(...)):
    return {"dates": stats_dates(sites.split(","))}


@router.get("/stats/data")
def stats_data(
    sites: str = Query(...),
    start_date: str = Query(...),
    end_date: str = Query(...),
):
    return get_stats_data(sites.split(","), start_date, end_date)

File: frontend/src/api/multi.ts
import { apiClient } from './client'

export async function fetchComparisonDates(sites: string[]) {
  const { data } = await apiClient.get('/api/multi/comparison/dates', {
    params: { sites: sites.join(',') },
  })
  return data.dates as string[]
}

export async function fetchComparisonData(sites: string[], date: string) {
  const { data } = await apiClient.get('/api/multi/comparison/data', {
    params: { sites: sites.join(','), date },
  })
  return data
}

export async function fetchIntervalDates(sites: string[]) {
  const { data } = await apiClient.get('/api/multi/interval/dates', {
    params: { sites: sites.join(',') },
  })
  return data.dates as string[]
}

export async function fetchIntervalData(sites: string[], startDate: string, endDate: string) {
  const { data } = await apiClient.get('/api/multi/interval/data', {
    params: { sites: sites.join(','), start_date: startDate, end_date: endDate },
  })
  return data
}

export async function fetchStatsDates(sites: string[]) {
  const { data } = await apiClient.get('/api/multi/stats/dates', {
    params: { sites: sites.join(',') },
  })
  return data.dates as string[]
}

export async function fetchStatsData(sites: string[], startDate: string, endDate: string) {
  const { data } = await apiClient.get('/api/multi/stats/data', {
    params: { sites: sites.join(','), start_date: startDate, end_date: endDate },
  })
  return data
}
